pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceTest.java

```diff
@@ -234,6 +237,56 @@ public void testBrokerServicePersistentTopicStats() throws Exception {
         assertEquals(subStats.msgBacklog, 0);
     }
 
+    @Test
+    public void testConnectionLimit() throws Exception {
+        cleanup();
+        conf.setBrokerMaxConnections(3);
```

Comment:
We should check corner cases 
brokerMaxConnections=1
brokerMaxConnectionsPerIp=0

brokerMaxConnections=0
brokerMaxConnectionsPerIp=1

brokerMaxConnections=0
brokerMaxConnectionsPerIp=0

brokerMaxConnections=1
brokerMaxConnectionsPerIp=1